src/main/scala/org/broadinstitute/hail/check/Gen.scala

```diff
@@ -114,13 +116,26 @@ object Gen {
       s.filter(_ => p.rng.getRandomGenerator.nextDouble <= cutoff))
   }
 
-  def sequence[C, T](gs: Traversable[Gen[T]])(implicit cbf: CanBuildFrom[Nothing, T, C]): Gen[C] =
+  def sequence[C[_], T](gs: Traversable[Gen[T]])(implicit cbf: CanBuildFrom[Nothing, T, C[T]]): Gen[C[T]] =
     Gen { (p: Parameters) =>
       val b = cbf()
       gs.foreach { g => b += g(p) }
       b.result()
     }
 
+  /**
+    * In general, for any Traversable type T and any Monad M, we may convert an {@code F[M[T]]} to an {@code M[F[T]]} by
+    * choosing to perform the actions in the order defined by the traversable. With {@code Gen} we must also consider
+    * the distribution of size. {@code uniformSequence} distributes the size uniformly across all elements of the
+    * traversable.
+    *
+    **/
+  def uniformSequence[C[_], T](gs : Traversable[Gen[T]])(implicit cbf: CanBuildFrom[Nothing, T, C[T]]): Gen[C[T]] =
+    getPartition(gs.size) map (resizeMany(gs, _)) flatMap sequence[C,T]
```

Comment:
We've been using the `.` style for method calls instead of juxtaposition.  I prefer stylistic consistency.  If you want to advocate expanded/different stylistic conventions, let's make that a separate conversation.